api/pkg/apis/v1alpha1/vendors/campaigns-vendor.go

```diff
@@ -82,30 +88,39 @@ func (c *CampaignsVendor) onCampaigns(request v1alpha2.COARequest) v1alpha2.COAR
 		namespace = "default"
 	}
 
+	version := request.Parameters["__version"]
+	rootResource := request.Parameters["__name"]
+	var id string
+	var resourceId string
+	if version != "" {
+		id = rootResource + "-" + version
+		resourceId = rootResource + ":" + version
+	} else {
+		id = rootResource
+		resourceId = rootResource
+	}
+	cLog.Infof("V (Campaigns): onCampaigns, id: %s, version: %s", id, version)
+
 	switch request.Method {
 	case fasthttp.MethodGet:
 		ctx, span := observability.StartSpan("onCampaigns-GET", pCtx, nil)
-		id := request.Parameters["__name"]
 		var err error
 		var state interface{}
-		isArray := false
-		if id == "" {
-			if !namespaceSupplied {
-				namespace = ""
-			}
-			state, err = c.CampaignsManager.ListState(ctx, namespace)
-			isArray = true
+
+		if version == "latest" {
```

Comment:
I have two qustions here:
1. what if version is empty? looks like the behavior is not correct
2. do we want to move latest check to manager layer?

Reply:
1. GetState(ctx, id, namespace) if version is empty. Id as name
2. Added GetLatestState to get for latest. Seems no need to combine all Get to GetState.

Reply:
what if version is empty and name doesn't contain the version

Reply:
Go to else block below: `state, err = c.CampaignsManager.GetState(ctx, id, namespace)`